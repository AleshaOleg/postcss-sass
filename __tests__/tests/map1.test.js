const getPostCssTreeFromSass = require('../../helpers/getPostCssTreeFromSassTree');

it('map1.sass', function () {
    expect(getPostCssTreeFromSass('map1')).toMatchSnapshot();
});
