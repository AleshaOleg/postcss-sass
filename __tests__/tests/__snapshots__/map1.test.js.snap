// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map1.sass 1`] = `
Object {
  "nodes": Array [
    Object {
      "prop": "$array",
      "raws": Object {
        "before": "",
        "between": ": ",
        "semicolon": false,
      },
      "source": Object {
        "end": Object {
          "column": 32,
          "line": 1,
        },
        "input": Input {
          "css": "$array: (1: '1', 2: '2', 3: '3')
$second_array: (4: '4', 5: '5')
$merged: map-merge($array, $second_array)

@each $num, $str in $merged
    .link-#{$str}
        z-index: $num
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 1,
        },
      },
      "type": "decl",
      "value": "(1: '1', 2: '2', 3: '3')",
    },
    Object {
      "prop": "$second_array",
      "raws": Object {
        "before": "
",
        "between": ": ",
        "semicolon": false,
      },
      "source": Object {
        "end": Object {
          "column": 31,
          "line": 2,
        },
        "input": Input {
          "css": "$array: (1: '1', 2: '2', 3: '3')
$second_array: (4: '4', 5: '5')
$merged: map-merge($array, $second_array)

@each $num, $str in $merged
    .link-#{$str}
        z-index: $num
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 2,
        },
      },
      "type": "decl",
      "value": "(4: '4', 5: '5')",
    },
    Object {
      "prop": "$merged",
      "raws": Object {
        "before": "
",
        "between": ": ",
        "semicolon": false,
      },
      "source": Object {
        "end": Object {
          "column": 41,
          "line": 3,
        },
        "input": Input {
          "css": "$array: (1: '1', 2: '2', 3: '3')
$second_array: (4: '4', 5: '5')
$merged: map-merge($array, $second_array)

@each $num, $str in $merged
    .link-#{$str}
        z-index: $num
",
          "id": "<input css 1>",
        },
        "start": Object {
          "column": 1,
          "line": 3,
        },
      },
      "type": "decl",
      "value": "map-merge($array, $second_array)",
    },
    Object {
      "nodes": Array [
        Object {
          "nodes": Array [
            Object {
              "prop": "z-index",
              "raws": Object {
                "before": "
        ",
                "between": ": ",
                "semicolon": false,
              },
              "source": Object {
                "end": Object {
                  "column": 21,
                  "line": 7,
                },
                "input": Input {
                  "css": "$array: (1: '1', 2: '2', 3: '3')
$second_array: (4: '4', 5: '5')
$merged: map-merge($array, $second_array)

@each $num, $str in $merged
    .link-#{$str}
        z-index: $num
",
                  "id": "<input css 1>",
                },
                "start": Object {
                  "column": 9,
                  "line": 7,
                },
              },
              "type": "decl",
              "value": "$num",
            },
          ],
          "raws": Object {
            "before": "
    ",
            "between": "",
          },
          "selector": ".link-#{str}",
          "source": Object {
            "end": Object {
              "column": 22,
              "line": 7,
            },
            "input": Input {
              "css": "$array: (1: '1', 2: '2', 3: '3')
$second_array: (4: '4', 5: '5')
$merged: map-merge($array, $second_array)

@each $num, $str in $merged
    .link-#{$str}
        z-index: $num
",
              "id": "<input css 1>",
            },
            "start": Object {
              "column": 5,
              "line": 6,
            },
          },
          "type": "rule",
        },
      ],
      "raws": Object {
        "before": "

",
        "between": "",
      },
      "selector": "@each $num, $str in $merged",
      "type": "rule",
    },
  ],
  "raws": Object {
    "before": "",
    "semicolon": undefined,
  },
  "source": Object {
    "end": Object {
      "column": 22,
      "line": 7,
    },
    "input": Input {
      "css": "$array: (1: '1', 2: '2', 3: '3')
$second_array: (4: '4', 5: '5')
$merged: map-merge($array, $second_array)

@each $num, $str in $merged
    .link-#{$str}
        z-index: $num
",
      "id": "<input css 1>",
    },
    "start": Object {
      "column": 1,
      "line": 1,
    },
  },
  "type": "root",
}
`;
